src/components/TaskNavBar/TaskNavBar.jsx

```diff
@@ -13,8 +21,14 @@ function TaskNavBar({ onCreateButtonClick }) {
         </button>
 
         <div className="task-navbar__filters">
-          {FILTER_TASKS_BUTTON_TEXT_ARRAY.map((filterButtonText, index) => (
-            <FilterButton buttonText={filterButtonText} key={index} />
+          {FILTER_TASKS_BUTTON_LABELS.map((filterButtonText, index) => (
+            <FilterButton
+              buttonText={filterButtonText}
+              key={index}
+              onFilter={() =>
+                handleFilterButtonClick(FILTER_TASKS_BUTTON_LABELS[index])
+              }
```

Comment:
Can we handle this without arrow function?

Reply:
@iud-wtag no vai we can't. 
The one you are suggesting will cause the `handleFilterButtonClick` to execute immediately during the rendering process, instead of being executed at `onFilter` event trigger. Which means, `onFilter` prop will not receive any function, rather it will get the immediate return value of `handleFilterButtonClick` which can be `undefined` during rendering.